src/s2s/eartts/sampler.cpp

```
@@ -0,0 +1,642 @@
+    sample_session_->run(inputs, outputs);
+    for (int b = 0; b < BT; ++b)
+        for (int q = 0; q < Q; ++q)
+            out_codes[static_cast<size_t>(b) * Q + q] = codes_qb[static_cast<size_t>(q) * BT + b];
+}
+
+
+// Host-orchestrated MaskGIT loop (any BT). Faithful port of tts_ggml's
+// run_maskgit + run_mog_step host-side parts: depth-sum, TopPLogitsWarper +
+// Gumbel-Max, low_mat matmul, z noise, per-quantizer RVQ nearest-neighbour.
+void
+EarTTSSampler::run_maskgit(
```

Clang (14.0.6)

[note] the definition seen here

(clang)